core/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -407,13 +408,23 @@ public CreateIndexRequest source(Map<String, ?> source) {
                     } catch (IOException e) {
                         throw new ElasticsearchParseException("failed to parse custom metadata for [{}]", name);
                     }
+                } else {
+                    // found a key which is neither custom defined nor one of the supported ones
+                    if (unsupportedKey == null) {
```

Comment:
@clintongormley I am not sure what you mean by 'deprecation logging'. Throw with a msg that this is no longer supported?